src/apps/basic/basic_apps.lua

```diff
@@ -86,6 +87,90 @@ function Sink:push ()
    end
 end
 
+--- ### `Match` app: Compare packets recieved on rx and comparator
+Match = {}
+
+function Match:new(cfg)
+   if cfg == "nil" then
+      cfg = { mode = "exact" }
+   end
+   assert(cfg.mode == "exact" or cfg.mode == "monotonic",
+      ("Unsupported mode %s"):format(cfg.mode))
+   return setmetatable({
+         cfg = cfg,
+         seen = 0,
+         errs = { },
+         done = false
+      },
+      {
+         __index=Match
+      })
+end
```

Comment:
Actually, the above breaks when `arg` is an empty table, this would be more correct:

```lua
function App:new (arg)
      local conf = config.parse_app_arg(arg) or {}
      conf.foo = conf.foo or "default"
end
```